src/audio/kpb.c

```diff
@@ -119,11 +123,212 @@ static void kpb_free(struct comp_dev *dev)
 	rfree(dev);
 }
 
+static int kpb_trigger(struct comp_dev *dev, int cmd)
+{
+	trace_kpb("kpb_trigger()");
+
+	return comp_set_state(dev, cmd);
+}
+
+/**
+ * kpb_prepare() - prepare key phrase buffer.
+ * @arg1:  kpb component.
+ *
+ * Return: integer representing either 0 - success
+ * or -EINVAL - failure.
+ */
+static int kpb_prepare(struct comp_dev *dev)
+{
+	struct comp_data *cd = comp_get_drvdata(dev);
+	int ret = 0;
+	int i;
+
+	trace_kpb("kpb_prepare()");
+
+	ret = comp_set_state(dev, COMP_TRIGGER_PREPARE);
+	if (ret < 0)
+		return ret;
+
+	cd->no_of_clients = 0;
+
+	/* allocate history_buffer/s */
+#if KPB_NO_OF_HISTORY_BUFFERS == 2
```

Comment:
Iirc, history buffers came from topology ?

Reply:
Well no, according to architecture design history buffer is internal to KPB.

Reply:
The rationale to keep history buffer as internal property of the component is that we want to optimize usage of HW resources and the component can hide internal logic to handle 2 physical buffers located in LP and HP SRAM.

Reply:
oh, ok -  had the recollection of seeing something like history buffers in topology message in topology but having this defined in Kconfig is fine too.